block/manager.go

```diff
@@ -249,6 +243,56 @@ func (m *Manager) AggregationLoop(ctx context.Context, lazy bool) {
 	}
 }
 
+func (m *Manager) SetFraudProofService(fraudProofServ *fraudserv.ProofService) {
+	m.executor.SetFraudProofService(fraudProofServ)
+}
+
+func (m *Manager) ProcessFraudProof(fraudProofService *fraudserv.ProofService, ctx context.Context, cancel context.CancelFunc) {
+	// subscribe to state fraud proof
+	sub, err := fraudProofService.Subscribe(types.StateFraudProofType)
+	if err != nil {
+		m.logger.Error("failed to subscribe to fraud proof gossip", "error", err)
+		return
+	}
+	defer sub.Cancel()
+
+	// continuously process the fraud proofs received via subscription
+	for {
```

Comment:
where is this for loop checking for the ctx shutting down or erroring out?
if it is relying on one of the call methods to error out if the ctx is ended I think that implicit complexity should either be clearly stated in a comment or we should be explicitly checking for the ctx end with a select statement at the top of the for loop. 

is the `sub.Proof` call a blocking call that only returns on proof recieved or context ended? if so then that is the comment I think we should add so that it is immediately clear to developers how this loop is being controlled.